alkira/resource_alkira_service_f5_lb.go

```diff
@@ -0,0 +1,362 @@
+		CreateContext: resourceF5LoadBalancerCreate,
+		ReadContext:   resourceF5LoadBalancerRead,
+		UpdateContext: resourceF5LoadBalancerUpdate,
+		DeleteContext: resourceF5LoadBalancerDelete,
+		CustomizeDiff: func(ctx context.Context, d *schema.ResourceDiff, m interface{}) error {
+			client := m.(*alkira.AlkiraClient)
+
+			old, _ := d.GetChange("provision_state")
+
+			if client.Provision == true && old == "FAILED" {
+				d.SetNew("provision_state", "SUCCESS")
+			}
+
+			return nil
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Name of the service.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"description": {
+				Description: "Description of the service.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"cxp": {
+				Description: "CXP on which the service should be provisioned.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"size": {
+				Description: "Size of the service, one of" +
+					" `SMALL`, `MEDIUM`, `LARGE`" +
+					" `2LARGE`, `5LARGE`.",
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"segment_ids": {
+				Description: "IDs of segments associated with the service.",
+				Type:        schema.TypeSet,
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"billing_tag_ids": {
+				Description: "IDs of billing tags to associate with" +
+					" the service.",
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeInt},
+				Optional: true,
+			},
+			"global_cidr_list_id": {
+				Description: "ID of global CIDR list from which subnets" +
+					" will be allocated for the external network interfaces of" +
+					" instances. These interfaces host the public IP" +
+					" addresses needed for virtual IPs.",
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"prefix_list_id": {
+				Description: "ID of prefix list to use for IP allowlist",
+				Type:        schema.TypeInt,
+				Optional:    true,
+			},
+			"segment_options": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Description: "The segment options as used by your F5 Load Balancer.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"segment_id": {
+							Description: "ID of the segment.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"nat_pool_prefix_length": {
+							Description: "Prefix length of subnets for the segment.",
+							Type:        schema.TypeInt,
+							Optional:    true,
+						},
+						"elb_nic_count": {
+							Description: "Number of nics to allocate for the segment.",
+							Type:        schema.TypeInt,
+							Required:    true,
+						},
+					},
+				},
+			},
+			"service_group_name": {
+				Description: "Name of the service group to be associated with the service.",
```

Comment:
I am not sure if we have a `service_group` concept. It's just group on portal? All those groups are actually just `group`. Could you double-check this one?

Reply:
From Cam : "Service groups are shown separately from user configured groups on the portal, but they are all fundamentally the same."

Reply:
so we don't have a `service_group`  resource in TF, it's just group. I would assume that it's the same group? Then why just call this `group`. Then it could be created by `alkira_group`? Does `service_group` have anything special?

Reply:
if it's created automatically, why it's a required field here...

Reply:
I actually don't know , I'm assuming just for naming or reference ?

Reply:
We also use service_group_name in INFOBLOX and PAN.
https://github.com/alkiranet/terraform-provider-alkira/blob/896ecb663c206276b4823e1bfbaf198fa3e712f1/alkira/resource_alkira_service_pan.go#L138
https://github.com/alkiranet/terraform-provider-alkira/blob/896ecb663c206276b4823e1bfbaf198fa3e712f1/alkira/resource_alkira_service_infoblox.go#L223
While the group is created automatically, we still want the name to be provided by the customer

Reply:
oh, well @AK-Benjamin, for those two, same issue as well. Our group name needs to be unique globally, so meaning, service group name needs to be unique? How it's supposed to work. The field was there, but it doesn't mean that it's all good. It's always confusing. I would tweak them to be just "service's group".